### PR TITLE
feat(components): add status indicator

### PR DESCRIPTION
**Pull request by james-sann**
## Checklist

- [ ] :link: considered opportunities for adding cross-reference URLs (grep for keywords)
- [ ] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [ ] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

File: src/pages/components/status-indicator.mdx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  status.
+package: '@zendeskgarden/react-avatars'
+components:
+  - avatars/src/elements/StatusIndicator.tsx
+---
+
+<Usage>
+  <Use>
+    <ul>
+      <li>To display contextual information for a users current status</li>
```
Comment (jzempel) [inline src/pages/components/status-indicator.mdx:14]:
```suggestion
      <li>To display contextual information for a user's current status</li>
```

File: src/pages/components/status-indicator.mdx [hunk 2]
```diff
@@ -0,0 +1,101 @@
+---
+title: Status indicator
+description: >
+  The Status indicator provides a consistent, accessible representation of a users current
```
Comment (jzempel) [inline src/pages/components/status-indicator.mdx:4]:
```suggestion
  The Status indicator provides a consistent, accessible representation of a user's current 
```

File: src/pages/components/status-indicator.mdx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+The Status indicator component follows this structure:
+
+<!-- prettier-ignore -->
+```jsx
+<StatusIndicator>
+  {/* Status text */}
+</StatusIndicator>
+```
+
+### Status indicator
```
Comment (jzempel) [inline src/pages/components/status-indicator.mdx:87]:
```suggestion
### StatusIndicator
```

This needs to match the code component name.

File: src/pages/components/status-indicator.mdx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+Provide an `aria-label` to ensure it is accessible.
+
+<Component components={props.data.mdx.components} componentName="statusIndicator" />
```
Comment (jzempel) [inline src/pages/components/status-indicator.mdx:91]:
```suggestion
<Component components={props.data.mdx.components} componentName="statusIndicator" />

Provide an `aria-label` to ensure it is accessible.
```

**Review by jzempel — CHANGES_REQUESTED** [review]
This is coming along nicely, @james-sann

File: src/pages/components/status-indicator.mdx (hunk 4 above)
Comment (jzempel) [reply src/pages/components/status-indicator.mdx:91]: fwiw https://github.com/zendeskgarden/react-components/pull/1413 will eventually fix the missing `<Component>` extends clause – which should always be at the top.

File: src/examples/avatar/StatusIndicatorType.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React from 'react';
+import { StatusIndicator } from '@zendeskgarden/react-avatars';
+import { Row, Col } from '@zendeskgarden/react-grid';
+
+const Example = () => (
```
[the remark this answers is not in the document]
Comment (james-sann) [reply src/examples/avatar/StatusIndicatorType.tsx:20]: Thanks @jzempel! I made an attempt that seems to work. Feel free to re-adjust if it needs tweaking.

**Comment by james-sann** [comment]
@jzempel All changes have been made.

File: src/examples/avatar/StatusIndicatorType.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (jzempel) [reply src/examples/avatar/StatusIndicatorType.tsx:20]: Perfect.

**Review by jzempel — APPROVED** [review]

**Pull request merged**